src/ufo.rs

```diff
@@ -194,6 +197,30 @@ impl Ufo {
                 .collect();
             let layers = layers?;
 
+            match meta.format_version {
```

Comment:
another little rust style thing, but is there a way we could write this that uses the match to avoid the `mut` annotations? For instance,

```rust
let (groups, kerning) = match (meta.format_version, groups, kerning) {
    // for v3, we do nothing
    (FormatVersion::V3, g, k) => (g, k),
    (_, None, k) => (None, k),
    (_, Some(g), k) => {
        let (groups, kerning) = upconvert_kerning(g, &k.unwrap_or_default());
        validate_groups(&groups)..maperr(Error::GroupsUpconversionError)?;
        (Some(groups), Some(kerning))
    }
}
```